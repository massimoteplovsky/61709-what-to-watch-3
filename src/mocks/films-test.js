export const films = [
  {
    id: 1,
    title: `Terminator 2. Judgment day`,
    genre: [`Action`],
    year: 1991,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 2,
    title: `1+1`,
    genre: [`Drama`, `Comedy`],
    year: 2012,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 3,
    title: `Joker`,
    genre: [`Drama`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 4,
    title: `Once upon a time in Hollywood`,
    genre: [`Drama`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 5,
    title: `Ford vs Ferrari`,
    genre: [`Drama`, `History`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 6,
    title: `1917`,
    genre: [`Drama`, `History`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 7,
    title: `Parasite`,
    genre: [`Drama`, `Comedy`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 8,
    title: `Irishman`,
    genre: [`drama`, `criminal`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  }
];

export const film = {
  id: 1,
  title: `Terminator 2. Judgment day`,
  genre: [`Action`],
  year: 1991,
  promoPoster: ``,
  poster: ``,
  cover: ``,
  description: ``,
  rating: 8.5,
  votes: 12000,
  director: ``,
  starring: []
};

export const promoFilmInfo = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  year: 2014
};
