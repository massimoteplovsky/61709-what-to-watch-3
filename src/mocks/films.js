export default [
  {
    id: 1,
    title: `Terminator 2. Judgment day`,
    poster: `http://i4.ytimg.com/vi/HgV7-MJwUBw/mqdefault.jpg`
  },
  {
    id: 2,
    title: `1+1`,
    poster: `https://i.pinimg.com/736x/76/44/c0/7644c07844dd4cb28d2c289bfa6e0dec--the-intouchables-comedy-movies.jpg`
  },
  {
    id: 3,
    title: `Joker`,
    poster: `https://all.culture.ru/uploads/9b743d135092e110a8e55a3dc3ad5bc4_w375_h250.jpeg`
  },
  {
    id: 4,
    title: `Once upon a time in Hollywood`,
    poster: `https://avatars.mds.yandex.net/get-pdb/1945878/3281b62e-4539-4eb7-a26b-fa68f6572bea/s375`
  },
  {
    id: 5,
    title: `Ford vs Ferrari`,
    poster: `https://sun9-28.userapi.com/c206816/v206816832/1771c/3p_wj8sfMz8.jpg`
  },
  {
    id: 6,
    title: `1917`,
    poster: `https://sun3-13.userapi.com/c857724/v857724955/14e882/u5CFMvbvX3I.jpg`
  },
  {
    id: 7,
    title: `Parasite`,
    poster: `http://m.kino-teatr.ru/art/5395/pv_75950.jpg`
  },
  {
    id: 8,
    title: `Irishman`,
    poster: `https://sun9-31.userapi.com/c850608/v850608676/40eb4/ojB5App6exk.jpg`
  },
];
