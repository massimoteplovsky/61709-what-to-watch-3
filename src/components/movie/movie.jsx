import React, {PureComponent} from 'react';
import {PropValidator} from '../../prop-validator/prop-validator';
import {connect} from 'react-redux';
import Header from '../header/header.jsx';
import MoviePromo from '../movie-promo/movie-promo.jsx';
import Tabs from '../tabs/tabs.jsx';
import MovieList from '../movie-list/movie-list.jsx';
import Footer from '../footer/footer.jsx';
import withActiveItem from '../../hocs/with-active-item/with-active-item';
import {makeRelatedFilmsList, getFilm, getFilmReviews} from '../../selectors/films/films';
import {loadFilmReviews} from '../../actions/action-creators/films/films';

const WrappedTabs = withActiveItem(Tabs);

class Movie extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const {onLoadFilmReviews} = this.props;
    const {id} = this.props.match.params;
    onLoadFilmReviews(id);
  }

  componentDidUpdate(prevProps) {
    const {onLoadFilmReviews} = this.props;
    const {id} = this.props.match.params;

    if (prevProps.match.params.id !== id) {
      onLoadFilmReviews(id);
    }
  }

  render() {
    const {
      filmInfo,
      relatedFilms,
      filmReviews
    } = this.props;

    if (!filmInfo) {
      return null;
    }

    const {
      name,
      backgroundImage,
      backgroundColor,
      posterImage,
    } = filmInfo;

    return (
        <>
          <section style={{background: backgroundColor}} className="movie-card movie-card--full">
            <div className="movie-card__hero">
              <div className="movie-card__bg">
                <img src={backgroundImage} alt={name} />
              </div>

              <h1 className="visually-hidden">WTW</h1>

              <Header/>

              <MoviePromo filmInfo={filmInfo}/>
            </div>

            <div className="movie-card__wrap movie-card__translate-top">
              <div className="movie-card__info">
                <div className="movie-card__poster movie-card__poster--big">
                  <img src={posterImage} alt={name} width="218" height="327" />
                </div>

                <div className="movie-card__desc">
                  <WrappedTabs
                    filmInfo={filmInfo}
                    filmReviews={filmReviews}
                  />
                </div>
              </div>
            </div>
          </section>

          <div className="page-content">
            <section className="catalog catalog--like-this">
              <h2 className="catalog__title">More like this</h2>
              <MovieList
                films={relatedFilms}
              />
            </section>

            <Footer/>
          </div>
        </>
    );
  }
}

Movie.propTypes = {
  relatedFilms: PropValidator.FILMS,
  filmInfo: PropValidator.FILM_INFO,
  filmReviews: PropValidator.FILM_REVIEW,
  onLoadFilmReviews: PropValidator.ON_LOAD,
  match: PropValidator.MATCH,
};

const mapStateToProps = (state, ownProps) => {
  const {id} = ownProps.match.params;
  return {
    relatedFilms: makeRelatedFilmsList(state, id),
    filmInfo: getFilm(state, id),
    filmReviews: getFilmReviews(state)
  };
};

const mapDispatchToProps = (dispatch) => ({
  onLoadFilmReviews(filmID) {
    dispatch(loadFilmReviews(filmID));
  }
});

export {Movie};
export default connect(mapStateToProps, mapDispatchToProps)(Movie);
