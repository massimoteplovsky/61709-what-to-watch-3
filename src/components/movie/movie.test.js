import React from 'react';
import renderer from 'react-test-renderer';
import {film, films, filmReviews} from '../../mocks/films-test';
import {Movie} from './movie.jsx';
import {Provider} from 'react-redux';
import {Router} from "react-router-dom";
import history from "../../history";
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore([]);

let store = mockStore({
  films: {
    filteredFilms: films,
    films,
    promoFilm: null,
    filmCounter: 8,
    actualGenre: ``,
    reviews: []
  },
  application: {
    error: false
  },
  user: {
    authorizationStatus: `NO_AUTH`,
    userInfo: null
  }
});

it(`<Movie /> component renders correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <Movie
              filmInfo={film}
              relatedFilms={films}
              filmReviews={filmReviews}
              onLoadFilmReviews={() => {}}
              match={{params: {id: 1}}}
            />
          </Router>
        </Provider>, {
          createNodeMock: () => {
            return {};
          }}).toJSON();
  expect(tree).toMatchSnapshot();
});


