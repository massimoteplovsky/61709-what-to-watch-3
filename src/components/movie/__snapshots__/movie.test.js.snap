// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Movie /> component renders correctly 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "background": "",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Film 3"
          src=""
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            className="user-block__link"
            href="/login"
            onClick={[Function]}
          >
            Sign in
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Film 3
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama
            </span>
            <span
              className="movie-card__year"
            >
              2019
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/1"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="14"
                viewBox="0 0 18 14"
                width="18"
              >
                <use
                  xlinkHref="#in-list"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn movie-card__button"
              href="add-review.html"
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Film 3"
            height="327"
            src=""
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              4,5
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Normal
              </span>
              <span
                className="movie-rating__count"
              >
                 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              
            </p>
            <p
              className="movie-card__director"
            >
              <strong>
                Director: 
                
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: 
                
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
          >
            <video
              height="175"
              muted={true}
              poster=""
              src=""
              style={
                Object {
                  "objectFit": "cover",
                }
              }
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/film/1"
              onClick={[Function]}
            >
              Film 1
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
          >
            <video
              height="175"
              muted={true}
              poster=""
              src=""
              style={
                Object {
                  "objectFit": "cover",
                }
              }
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/film/2"
              onClick={[Function]}
            >
              Film 2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
          >
            <video
              height="175"
              muted={true}
              poster=""
              src=""
              style={
                Object {
                  "objectFit": "cover",
                }
              }
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/film/3"
              onClick={[Function]}
            >
              Film 3
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
